osidb/serializer.py

```diff
@@ -1453,6 +1456,16 @@ class Meta:
             + WorkflowModelSerializer.Meta.fields
         )
 
+    def create(self, validated_data):
+        component = validated_data.pop("component", None)
+        components = validated_data.pop("components", None)
+        if component and not components:
+            # Split the component and update the components to provide retro-compatibility
+            validated_data["components"] = [
+                value.strip() for value in component.split(":")
```

Comment:
Even though the split would have no impact (so would not be an issue) when there is only a single component, I would not expect the user to put multiple components where one is expected.

Reply:
makes sense, I will add the input string as a single value in `components`